src/lint/rules/subject-max-length.ts

```diff
@@ -0,0 +1,18 @@
+import { createRule, RuleFactoryCheckResults } from "./factory";
+import { Commit } from "@commitlint/parse";
+
+const maxLength = 50;
+
+export default createRule({
+  name: "subject-max-length",
+  score: 5,
+  check: (commit: Commit): RuleFactoryCheckResults => {
+    const actualLength = commit.header.length;
```

Comment:
JS support for Unicode is not too good. To get string length (i.e. number of Unicode code points) in JS, we should use `[...str].length` instead. `[...str]` would convert the string into an array of Unicode code points.